### PR TITLE
Introduce additional scanning capabilities to Qualys Scanner

### PR DESCRIPTION
**Pull request by mathuriga**
## Purpose
During manual Qualys Scan effort during recent pre-release scans, we have faced some issues and we observed the need for a change of default configurations of web applications. Therefore, We have done the Qualys gap analysis to make user usability much easier. 

The following are the items that we have decided to improve the current Qualys Scanner implementation of Scan Manager. This PR includes the implementation of this improvement.

- Support Standard Authentication in addition to Selenium Script authentication

- Provide users to provide Crawling scope for web app

          Supported values; ALL (default), LIMIT, SUBDOMAIN or DOMAINS.

            -  If set to ALL, the scan will crawl all directories and subdirectories of the starting URL.
            -  If set to LIMIT, crawling will be limited to the starting URI’s initial path and sub-directories.
            -  If set to SUBDOMAINS, any sub-domain that is in the same domain as the specified 
               domain name will be crawled.
            -  If set to DOMAINS, only the specified domains will be crawled

- Provide BlackList for crawling: Regex option

- Support for crawling script

## Goals
To allow users to perform above use cases while doing Qualys scan.

## Approach
Using available API in Qualys side, these capabilities were implemented.

## User stories
N/A

## Release note
N/A

## Documentation
N/A

## Training
N/A

## Certification
N/A

## Marketing
N/A

## Automation tests
N/A

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

## Samples
N/A

## Related PRs
N/A

## Migrations (if applicable)
N/A

## Test environment
N/A
 
## Learning
N/A

### CODE REVIEW

**Comment by ayomawdb** [comment]
Approving based on https://travis-ci.org/github/wso2/security-tools/builds/675152929

**Comment by ayomawdb** [comment]
Better to avoid word "some" in PR titles. It adds some uncertainty to the contribution IMO. In this repo it's fine, but letting you know so that you can think of it in future. Better title would be following IMO:

Introduce additional scanning capabilities to Qualys Scanner

File: internal/scan-manager/scanners/qualys/src/main/java/org/wso2/security/tools/scanmanger/qualys/auth/StandardAuthenticationRecordBuilder.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        webAppAuthRecord.appendChild(authRecordName);
+
+        Element formRecord = doc.createElement(QualysScannerConstants.FORM_RECORD);
+        webAppAuthRecord.appendChild(formRecord);
+
+        Element type = doc.createElement(QualysScannerConstants.TYPE_KEYWORD);
+        type.appendChild(doc.createTextNode(QualysScannerConstants.STANDARD_AUTH));
+        formRecord.appendChild(type);
+
+        //        Element sslOnly = doc.createElement(QualysScannerConstants.SSL_ONLY);
```
Comment (ayomawdb) [inline internal/scan-manager/scanners/qualys/src/main/java/org/wso2/security/tools/scanmanger/qualys/auth/StandardAuthenticationRecordBuilder.java:81]: Is the necessary? If you want to retain this, shall we add a comment explaining why this is commented. Also better to remove leading spaces.

File: internal/scan-manager/scanners/qualys/src/main/java/org/wso2/security/tools/scanmanager/scanners/qualys/model/ScanContext.java [hunk 2]
```diff
@@ -68,6 +74,52 @@
     // Report template id to generate report
     private String reportTemplateId;
 
+    // Authentication builder type
```
Comment (ayomawdb) [inline internal/scan-manager/scanners/qualys/src/main/java/org/wso2/security/tools/scanmanager/scanners/qualys/model/ScanContext.java:77]:
```suggestion
    // Authentication builder type.
```
Same with other lines.

File: internal/scan-manager/scanners/qualys/src/main/java/org/wso2/security/tools/scanmanager/scanners/qualys/model/SeleniumScript.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+import java.io.IOException;
+
+/**
+ * Model class to represent Selenium Script Object.
+ */
+public abstract class SeleniumScript {
+
+    private static final Logger log = LogManager.getLogger(SeleniumScript.class);
+
+    // Script File
```
Comment (ayomawdb) [inline internal/scan-manager/scanners/qualys/src/main/java/org/wso2/security/tools/scanmanager/scanners/qualys/model/SeleniumScript.java:45]:
```suggestion
    // Reference to the Selenium script.
```

File: internal/scan-manager/scanners/qualys/src/main/java/org/wso2/security/tools/scanmanager/scanners/qualys/model/SeleniumScript.java [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    private File scriptFile;
+
+    // Name of the script file
```
Comment (ayomawdb) [inline internal/scan-manager/scanners/qualys/src/main/java/org/wso2/security/tools/scanmanager/scanners/qualys/model/SeleniumScript.java:48]:
```suggestion
    // Name of the script file.
```

File: internal/scan-manager/scanners/common/src/main/java/org/wso2/security/tools/scanmanager/scanners/common/util/FileUtil.java [hunk 5]
```diff
@@ -95,8 +96,8 @@ public static String extractArchive(File sourceFile, String destination) throws
                 if (canonicalizedDestinationFilePath.startsWith(new File(destination).getCanonicalPath())) {
                     // if a valid zip file uploaded
                 } else {
-                    String errorMessage = "Attempt to upload invalid zip archive with file at " + currentEntry +
-                            ". File path is outside target directory";
+                    String errorMessage = "Attempt to upload invalid zip archive with file at " + currentEntry
+                            + ". File path is outside target directory";
```
[the remark this answers is not in the document]
Comment (mathuriga) [reply internal/scan-manager/scanners/common/src/main/java/org/wso2/security/tools/scanmanager/scanners/common/util/FileUtil.java:100]: Resolved.

**Pull request merged**